### ISSUE TITLE
VBT giving bad error at the very end of exporting big number of products

### ISSUE THREAD

**Issue by KevinGossentCap** [vlocityinc/vlocity_build#530]
VBT 1.16
nodeJS 18.12.1
Windows 10

Having 6300 products in catalogs with attributes and pricelistentries
exporting the whole catalog with all products
job ends with the following error
Uncaught Job Error >> RangeError: Invalid string length
    at JSON.stringify (<anonymous>)
    at DataPacksJob.exportFromManifest (C:\KGO_Tools\node-v18-win-x64\node_modules\vlocity\lib\datapacksjob.js:1200:49)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async DataPacksJob.exportJob (C:\KGO_Tools\node-v18-win-x64\node_modules\vlocity\lib\datapacksjob.js:1076:5)
    at async DataPacksJob.doRunJob (C:\KGO_Tools\node-v18-win-x64\node_modules\vlocity\lib\datapacksjob.js:333:9)
    at async DataPacksJob.runJobWithInfo (C:\KGO_Tools\node-v18-win-x64\node_modules\vlocity\lib\datapacksjob.js:293:9)
    at async DataPacksJob.runJob (C:\KGO_Tools\node-v18-win-x64\node_modules\vlocity\lib\datapacksjob.js:95:26)
    at async Vlocity.runDataPacksCommand (C:\KGO_Tools\node-v18-win-x64\node_modules\vlocity\lib\vlocity.js:173:16)
    at async VlocityCLI.runJob (C:\KGO_Tools\node-v18-win-x64\node_modules\vlocity\lib\vlocitycli.js:725:12)
    at async VlocityCLI.runCommands (C:\KGO_Tools\node-v18-win-x64\node_modules\vlocity\lib\vlocitycli.js:613:23) RangeError: Invalid string length
    at JSON.stringify (<anonymous>)
    at DataPacksJob.exportFromManifest (C:\KGO_Tools\node-v18-win-x64\node_modules\vlocity\lib\datapacksjob.js:1200:49)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async DataPacksJob.exportJob (C:\KGO_Tools\node-v18-win-x64\node_modules\vlocity\lib\datapacksjob.js:1076:5)
    at async DataPacksJob.doRunJob (C:\KGO_Tools\node-v18-win-x64\node_modules\vlocity\lib\datapacksjob.js:333:9)
    at async DataPacksJob.runJobWithInfo (C:\KGO_Tools\node-v18-win-x64\node_modules\vlocity\lib\datapacksjob.js:293:9)
    at async DataPacksJob.runJob (C:\KGO_Tools\node-v18-win-x64\node_modules\vlocity\lib\datapacksjob.js:95:26)
    at async Vlocity.runDataPacksCommand (C:\KGO_Tools\node-v18-win-x64\node_modules\vlocity\lib\vlocity.js:173:16)
    at async VlocityCLI.runJob (C:\KGO_Tools\node-v18-win-x64\node_modules\vlocity\lib\vlocitycli.js:725:12)
    at async VlocityCLI.runCommands (C:\KGO_Tools\node-v18-win-x64\node_modules\vlocity\lib\vlocitycli.js:613:23)

**Comment by manas-sf** [comment]
I think 6300 is a very large number and manipulating all of them somewhere as JSON, has hit the Javascript limits for performing JSON.stringify(). It is advisable to use small numbers for faster deployments and acknowledge success/error messages on the go. If needed, You may also add additional filters to query only relevant DataPacks.

**Comment by KevinGossentCap** [comment]
As a mater of fact I think I've discovered the issue.
There is a real issue here in fact.
I've finally managed to force catalog foldername to globalkey (something for which you really have to improve the documentation).
Doing that, I didn't have the issue.
what I've seen is that we have some catalogs with names that are almost identical (and if we apply both vlocity special characters suppression and Case insensible in windows they becomes identical).
What it give is that some catalogs are overriding others and in the end we end up with one of the described known issues [STRING_TOO_LONG, JSONAttribute](https://github.com/vlocityinc/vlocity_build#known-issues)

All that because of bad standard choice of foldername that could never be sure to be unique.
